test/src/tests-session-manager.ts

```diff
@@ -0,0 +1,860 @@
+
+                mpInstance._Store.isEnabled = false;
+
+                mpInstance._SessionManager.endSession();
+
+                // Should log initial StartingEndSession and AbandonEndSession messagesk
+                expect(consoleSpy.getCalls().length).to.equal(2);
+                expect(consoleSpy.lastCall.firstArg).to.equal(
+                    Messages.InformationMessages.AbandonEndSession
+                );
+            });
+
+            // QUESTION: What exactly defines an "AbandonedEndSession"?
+            // FIXME: Test only passes if specific cases are set
+            it.skip('should log an AbandonedEndSession message if webviewBridgeEnabled is not enabled', () => {
+                mParticle.init(apiKey, window.mParticle.config);
+
+                const mpInstance = mParticle.getInstance();
+                sinon
+                    .stub(mpInstance._Persistence, 'getPersistence')
+                    .returns({ gs: {} });
+
+                const consoleSpy = sinon.spy(mpInstance.Logger, 'verbose');
+
+                // FIXME: This test only passes if devToken is undefined
+                //        If it is set, webviewBridgeEnabled is ignored
+                mpInstance._Store.webviewBridgeEnabled = false;
+                mpInstance._Store.devToken = undefined;
+
+                mpInstance._SessionManager.endSession();
+
+                // Should log initial StartingEndSession and AbandonEndSession messagesk
+                expect(consoleSpy.getCalls().length).to.equal(2);
+                expect(consoleSpy.lastCall.firstArg).to.equal(
+                    Messages.InformationMessages.AbandonEndSession
+                );
+            });
+
+            // QUESTION: What exactly defines an "AbandonedEndSession"?
+            // FIXME: Test only passes if specific cases are set
+            it.skip('should log an AbandonedEndSession message if devToken is not set', () => {
+                mParticle.init(apiKey, window.mParticle.config);
+
+                const mpInstance = mParticle.getInstance();
+                sinon
+                    .stub(mpInstance._Persistence, 'getPersistence')
+                    .returns({ gs: {} });
+
+                const consoleSpy = sinon.spy(mpInstance.Logger, 'verbose');
+
+                // FIXME: Test only passes if devToken is undefined and webviewBridgeEnabled is false
+                mpInstance._Store.devToken = undefined;
+                mpInstance._Store.webviewBridgeEnabled = false;
+
+                mpInstance._SessionManager.endSession();
+
+                // Should log initial StartingEndSession and AbandonEndSession messagesk
+                expect(consoleSpy.getCalls().length).to.equal(2);
+                expect(consoleSpy.lastCall.firstArg).to.equal(
+                    Messages.InformationMessages.AbandonEndSession
+                );
+            });
+
+            it('should priortize cookie sessionId over Store.sessionId', () => {
```

Comment:
I think it's fine to keep this unit test, but I think in real life scenarios, these are always in sync.  If you found a case where it isn't, then that's a bug.